app/src/main/java/com/android/solvit/provider/ui/NotificationsScreen.kt

```diff
@@ -30,32 +52,66 @@ fun NotificationScreen(
   LaunchedEffect(providerId) { viewModel.init(providerId) }
 
   val notifications by viewModel.notifications.collectAsState()
+  var showDialog by remember { mutableStateOf(false) }
+  val selectedServiceRequest = remember { mutableStateOf<ServiceRequest?>(null) }
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = {
+              Row(
+                  modifier = Modifier.fillMaxWidth().testTag("notifications_title"),
+                  horizontalArrangement = Arrangement.Start) {
+                    Text("Notifications", textAlign = TextAlign.Center)
+                  }
+            },
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.goBack() },
+                  modifier = Modifier.testTag("goBackButton")) {
+                    Icon(Icons.Filled.ArrowBack, contentDescription = "Back")
+                  }
+            })
+      }) { paddingValues ->
+        // Content inside the Scaffold
+        Column(modifier = Modifier.padding(paddingValues)) {
+          if (notifications.isEmpty()) {
+            Text(
+                text = "No notifications available",
+                modifier = Modifier.fillMaxSize().testTag("noNotificationsText"), // Added test tag
+                textAlign = TextAlign.Center,
+                style = MaterialTheme.typography.bodySmall)
+          } else {
+            LazyColumn(
```

Comment:
Maybe here to handle scalability integrate lazy loading/paging using LazyPagingItems from the Paging library, its a small fix.

Reply:
Good suggestion! This will ensure better performance as the list grows.Thanks for the tip!